### ISSUE TITLE
Text To Speech to Facial BlendShapes 

### ISSUE THREAD

**Issue by GeorgeS2019** [google/mediapipe#4428]
### MediaPipe Solution (you are using)

Face Blendshape

### Programming language

c#

### Are you willing to contribute it

Yes

### Describe the feature and the current behaviour/state

```shell
From the Modelling part using Godot
https://github.com/srcnalt/ReadyPlayerMe-Godot-Test/issues/1#issue-1713856035
```

### Will this change the current API? How?

```shell
YES
```

### Who will benefit with this feature?

Anyone who use MediaPipe BlendShape. It is NEXT STEP to Deep AI  (Integrating Deep Audio to MediaPipe)

### Please specify the use cases for this feature

```shell
User use ChatGPT or something similar to generate replies and this new feature translate the replies to speech with corresponding Avatar Blendshapes manipulation
```

**Comment by GeorgeS2019** [comment]
How the API looks Like ?

Given a ChatGPT or something similar from Google reply in text, the API will receive this string and output the corresponding facial blendshapes as Dictionary[ blendshapeName, blendshapeValueFloat]

**Comment by GeorgeS2019** [comment]
@endink 
This is just Part 2 of many parts ahead :-)

**Comment by FishWoWater** [comment]
Agreed! It would be really exciting if blendshapes could be estimated and aligned with input aduio clip.

I am currently working on a pipeline:  `user voice->speech recognition->chatgpt->text to speech->blendshapes`. There exist many mature solutions except for the last stage (speech2blendshapes). [Lipsync](https://developer.oculus.com/downloads/package/oculus-lipsync-unity/) and [face good](https://github.com/FACEGOOD/FACEGOOD-Audio2Face) can possibly do this, but have their limitations or problems. This feature will benefit the mediapipe community.

**Comment by ayushgdev** [comment]
Hello @GeorgeS2019 Thanks for raising this amazing feature request. We will discuss it internally and prioritise it in our roadmap. However, just a heads up, we are working in numerous fronts as of now hence this might get delayed.

**Comment by GeorgeS2019** [comment]
Now working, the BlendShape part in **8th Top Ranked Github Open source** 3D game engine: [Godot](https://github.com/j20001970/GDMP-demo)
@srcnalt 
@kaiidams 
@SpookyCorgi 
@you-win 
@j20001970
![Godot_v4 0 3-rc2_mono_win64_JU4OlmIfLZ](https://github.com/google/mediapipe/assets/49812372/6dd69fef-30a1-4713-a95e-c10a4493d885)

**Comment by kuaashish** [comment]
Hello @lu-wang-g,
Could you please look into this amazing feature request? Thank you!!

**Comment by lu-wang-g** [comment]
At I/O 2023, Google released the demo app, Talking Character (https://developers.googleblog.com/2023/05/generative-ai-talking-character.html), which IIUC fits exactly the use case described here. The Web demo is partially open sourced [here](https://github.com/masa8/generative-ai-docs/tree/131fcce6510e7762cb81c974ecc25ead6d4f26ca/demos/palm/web). You can find useful pieces of components in the directory. There has also been a discussion of releasing the talking character pipeline through MediaPipe, but we don't have concrete plan yet. 

@ayushgdev and @kuaashish, do we have ways to track user requests like this?

**Comment by GeorgeS2019** [comment]
@ayushgdev  @kuaashish @lu-wang-g

I went through the feedback. 

First, this user request is currently one of the hottest AI showcases of big tech giants, including Nvidia's [Audio2Face](https://www.nvidia.com/en-us/omniverse/apps/audio2face/).

Microsoft has an Azure solution for it and I went through it.

The implementation is ~80% compliant with the industry standard to the proposed Arkit 52 blendshapes recently adopted by Mediapipe.

I have seen users complain about the Azure closed source to Audio2Face.

Closed source means the lack of agility when compared to Nivida refinement environment


@lu-wang-g

The Talking Character you have cited. 
- First, it is a 2D flat avatar image, which immediately is MUCH less than that of Azure 3D avatar Audio2Face solution, a good start but much to go.
- Second, the existing Mediapipe Facial Landsmark (with blendshape) is a major and significant contribution to humanity. However, the facial Mocap always suffers from interpreting the facial mocap when the captured face is e.g. Half Facing the camera.

==> Audio2Face is disruptive to such Facial Mocap limitations

- Third, as I said, this is a major contribution to humanity, there are many innovative AI with many creative milestones ahead by following the path of Audio2Face instead of the conventional incumbent facial mocap.

I urge Google to start thinking about supporting top ranked  (8th of all GitHub projects) MIT open-source 3D engine communities like e.g. Godot.

Why? Just like how Android technology has democratized mobility communication. Google needs the most popular 3D game engine to align with the Android mission

Godot is fully capable of providing all the needed 3D requirements, compatible with industry giants like Unity and Unreal and soon Nividia.

Godot users are creating OpenVerse and photorealism 3D avatars with human-like movements, thanks to the next generation of deep reinforcement AI.

@lyuma  Who is one of the top contributors to Godot on 3D avatar in VR space can fill in the rest of why Google needs to seriously consider supporting Godot.

Why ME??

My mission is to disseminate AI-powered technology to all corners of the world to END the Long Covid crisis. The US will be facing the next generation's cunning and deceptive Omicron subvariant this Winter. I need your support to help me save thousands of lives.

Finally, I appreciate Google's support in considering my sincere request for the common good of making a difference to humanity.

**Comment by tiamy** [comment]
+1

**Comment by GeorgeS2019** [comment]
We now have C# wrapper of Godot Mediapipe

**Comment by GeorgeS2019** [comment]
The Godot community will attempt Text to Face => follow [here ](https://github.com/google/mediapipe/issues/5366)

**Comment by admingistai** [comment]
> At I/O 2023, Google released the demo app, Talking Character (https://developers.googleblog.com/2023/05/generative-ai-talking-character.html), which IIUC fits exactly the use case described here. The Web demo is partially open sourced [here](https://github.com/masa8/generative-ai-docs/tree/131fcce6510e7762cb81c974ecc25ead6d4f26ca/demos/palm/web). You can find useful pieces of components in the directory. There has also been a discussion of releasing the talking character pipeline through MediaPipe, but we don't have concrete plan yet.
> 
> [@ayushgdev](https://github.com/ayushgdev) and [@kuaashish](https://github.com/kuaashish), do we have ways to track user requests like this?

Is there any update on releasing the MediaPipe talking character pipeline? I have been looking for solutions for audio to lipsynced cartoon character just like in the demo, but the focus of models released since then seems to be on photorealistic video. I can't find anything quite like that demo.
